### ISSUE TITLE
USB3 Point Grey Chameleon3 CM3-U3-13S2C (colour) outputs no colour images (only grey).

### ISSUE THREAD

**Issue by Crusty82** [ros-drivers/pointgrey_camera_driver#63]
Model: Chameleon3 CM3-U3-13S2C
ROS: Indigo
Machine: 64 Bit Intel
OS: 64Bit Ubuntu 14.04.1

Further context here: http://answers.ros.org/question/231635/ros-point-grey-chameleon-usb3-cam-no-colour-images/

Temporarily resolved by switching to Kumar Robotics' ROS stack for Point Grey USB3 Cameras, which works for me.

**Comment by m91moreira** [comment]
Hi,

I'm having the same problem with the same camera model.

I tryed to use the Kumar Robotics stack, but I'm only abble to get no color images. I've tryed all the video modes without success...

What video mode did you choose?

**Comment by Crusty82** [comment]
Video Mode 23.

**Comment by m91moreira** [comment]
I've tried before and it didn't work.

Did you change your launch file?

**Comment by Crusty82** [comment]
No, I didn't change the launch file. What topic are you looking at when visualizing the image?

**Comment by m91moreira** [comment]
After running the launch file, I run the image_proc, in order to get the color images, but I get a warning saying " Color topic '/pg_15374753/image_color' requested, but raw image data from topic '/pg_15374753/image_raw' is grayscale".

I'm trying to look to /image_color topic.

**Comment by m91moreira** [comment]
It seems that the problem is in the flycapture lib.

It doesn't get the isColorCamera field correctly on armhf. In order to get color on the odroid, I forced all the verifications of isColorCamera to true.

**Comment by jgurzoni** [comment]
This issue seem to have been resolved with the library compilation against the 2.9.3.43 library

**Comment by wynn4** [comment]
I'm having the same problem using Chameleon3, Ubuntu 16.04 LTS, Odroid XU4, and flycapture.2.11.3.121_armhf.  The only place I can see to change the isColorCamera flag is in flycapture.2.11.3.121_armhf/include/FlyCapture2Defs.h.  Setting it to true and I'm still only getting a mono8 grayscale image. Am I missing something?  rosrun pointgrey_camera_driver list_cameras still says that Color is false

**Comment by IanBoyanZhang** [comment]
Hi @wynn4, were you able to fix the issue? We are having similar issues with "Point Grey Grasshopper 3". Thanks!

**Comment by wynn4** [comment]
@IanBoyanZhang, are you also using an ARM processor such as the Odroid XU4 or TX1?  Have you installed it via apt-get, or have you built it from source.  In our case, the color image didn't work when using apt-get, but it did work when we built it from source.  You may also have to grep for 'isColorCamera' in the flycapture lib and force it to be true.  When building from source, there is a header file that won't be found (flycapture2.h I think) but all you have to do is add the location of that file as an include directory in the CMakeLists.txt.  Hope that helps.

**Comment by IanBoyanZhang** [comment]
@wynn4 Thanks for getting back to me! We are running it on PC instead of an ARM SoC. We fixed the CMakeLists. It resolves the problem. Thanks again.
